src/libopensc/pkcs15-esteid2025.c

```diff
@@ -182,12 +233,12 @@ sc_pkcs15emu_esteid2025_init(sc_pkcs15_card_t *p15card)
 		struct sc_pkcs15_prkey_info prkey_info = {
 				.id = {.len = 1, .value[0] = prkey_id[i]},
 				.native = 1,
-				.key_reference = prkey_ref[i],
+				.key_reference = prkey_ref[is_fin_eid(p15card)][i],
 				.field_length = field_length,
 				.usage = prkey_usage[i]
 		       };
 		struct sc_pkcs15_object prkey_obj = {
-				.auth_id = {.len = 1, .value[0] = prkey_pin[i]},
+				.auth_id = {.len = 1, .value[0] = pin_authid[i]},
```

Comment:
just a nit that this mapping is really just mapping `i` -> `i + 1`. But given that it is used at least at two places, its probably ok to use it as it is.